clients/urlclient/registry_test.go

```diff
@@ -55,11 +47,15 @@ func TestRegistryClient_URLPrefix(t *testing.T) {
 
 func TestRegistryClient_URLPrefixInitialized(t *testing.T) {
 	expectedURL := "http://domain.com"
-	client := newRegistryClient(types.EndpointParams{}, mockEndpoint{}, 100)
-	client.initialized = true
-	client.url = expectedURL
+	urlClient := newRegistryClient(types.EndpointParams{}, mockEndpoint{}, 100)
+
+	// set up prerequisite condition
+	actualURL, err := urlClient.Prefix()
+	if err != nil {
+		t.Fatalf("unexpected error in precondition %s", err.Error())
+	}
```

Comment:
This implementation duplicates the implementation that immediately follows it.

Reply:
I'm trying to ensure that initialized url retrieval (which is a different code path) works as expected. Do you have any suggestions on how to better accomplish that without modifying the internal state?

Reply:
`TestRegistryClient_URLPrefix` may not actually test what you expect.  Behavior is indeterminate and dependent on timing.  In one execution, it may exercise the `c.Initialized` path in `Prefix()`.  In another, it may fall into the for loop.

I'd replace the interfaces.Endpointer mock with a stub that creates the channel returned by Monitor when the stub is constructed.  I'd add a `SendToChannel()` method to the stub that allows the test to control the timing of the monitor's population of the `registryClient`'s URL property.  I'd then use the new method in both `TestNewRegistryClient` and `TestRegistryClient_URLPrefix` to facilitate the desired tests.  I'd include comments within the tests that document the intent behind the specific testing implementations.

As an aside, the goroutine in `mockTimeoutEndpoint.Monitor()` is superfluous and should be removed.

Reply:
To be handled in follow up issue https://github.com/edgexfoundry/go-mod-core-contracts/issues/204